libz-rs-sys/src/gz.rs

```diff
@@ -0,0 +1,239 @@
+use zlib_rs::allocate::*;
+pub use zlib_rs::c_api::*;
+
+use core::ffi::{c_char, c_int, c_uint, CStr };
+use core::mem::size_of;
+use core::ptr;
+use libc::{O_APPEND, O_CREAT, O_EXCL, O_RDONLY, O_TRUNC, O_WRONLY, SEEK_CUR, SEEK_END};
+use zlib_rs::deflate::Strategy;
+
+/// For compatibility with the zlib C API, this structure exposes just enough of the
+/// internal state of an open gzFile to support the gzgetc() C macro.
+#[repr(C)]
+#[derive(Copy, Clone)]
+pub struct gzFile_s {
+    have: c_uint,        // number of bytes available at next
+    next: *const Bytef,  // next byte of uncompressed data
+    pos: u64,            // current offset in uncompressed data stream
+}
```

Comment:
I think we should instead make this type opaque to rust users, so similarly to https://docs.rs/libz-sys/1.1.22/libz_sys/enum.gzFile_s.html use an enum with no variants. The `GzFile` can just store these fields, with a comment that the fields must not move